File: rest/src/main/java/io/sparkled/viewmodel/playlist/sequence/PlaylistSequenceViewModel.java
package io.sparkled.viewmodel.playlist.sequence;

import java.util.Objects;
import java.util.UUID;

public class PlaylistSequenceViewModel {

    private UUID uuid;
    private Integer sequenceId;
    private Integer displayOrder;

    public UUID getUuid() {
        return uuid;
    }

    public PlaylistSequenceViewModel setUuid(UUID uuid) {
        this.uuid = uuid;
        return this;
    }

    public Integer getSequenceId() {
        return sequenceId;
    }

    public PlaylistSequenceViewModel setSequenceId(Integer sequenceId) {
        this.sequenceId = sequenceId;
        return this;
    }

    public Integer getDisplayOrder() {
        return displayOrder;
    }

    public PlaylistSequenceViewModel setDisplayOrder(Integer displayOrder) {
        this.displayOrder = displayOrder;
        return this;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        PlaylistSequenceViewModel that = (PlaylistSequenceViewModel) o;
        return Objects.equals(uuid, that.uuid) &&
                Objects.equals(sequenceId, that.sequenceId) &&
                Objects.equals(displayOrder, that.displayOrder);
    }

    @Override
    public int hashCode() {
        return Objects.hash(uuid, sequenceId, displayOrder);
    }

    @Override
    public String toString() {
        return "PlaylistSequenceViewModel{" +
                "uuid=" + uuid +
                ", sequenceId=" + sequenceId +
                ", displayOrder=" + displayOrder +
                '}';
    }
}

File: rest/src/main/kotlin/io/sparkled/viewmodel/playlist/sequence/PlaylistSequenceViewModel.kt
package io.sparkled.viewmodel.playlist.sequence

import io.sparkled.viewmodel.ViewModel
import java.util.UUID

class PlaylistSequenceViewModel : ViewModel {

    private var uuid: UUID? = null
    private var sequenceId: Int? = null
    private var displayOrder: Int? = null

    fun getUuid(): UUID? {
        return uuid
    }

    fun setUuid(uuid: UUID?): PlaylistSequenceViewModel {
        this.uuid = uuid
        return this
    }

    fun getSequenceId(): Int? {
        return sequenceId
    }

    fun setSequenceId(sequenceId: Int?): PlaylistSequenceViewModel {
        this.sequenceId = sequenceId
        return this
    }

    fun getDisplayOrder(): Int? {
        return displayOrder
    }

    fun setDisplayOrder(displayOrder: Int?): PlaylistSequenceViewModel {
        this.displayOrder = displayOrder
        return this
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as PlaylistSequenceViewModel

        if (uuid != other.uuid) return false
        if (sequenceId != other.sequenceId) return false
        if (displayOrder != other.displayOrder) return false

        return true
    }

    override fun hashCode(): Int {
        var result = uuid?.hashCode() ?: 0
        result = 31 * result + (sequenceId ?: 0)
        result = 31 * result + (displayOrder ?: 0)
        return result
    }

    override fun toString(): String {
        return "PlaylistSequenceViewModel(uuid=$uuid, sequenceId=$sequenceId, displayOrder=$displayOrder)"
    }
}
